src/set.rs

```diff
@@ -0,0 +1,117 @@
+use std::cell::{Cell, UnsafeCell};
+use std::collections::hash_map::{Entry, HashMap};
+use std::hash::Hash;
+use std::iter::FromIterator;
+
+use stable_deref_trait::StableDeref;
+
+/// Append-only version of `std::collections::HashSet` where
+/// insertion does not require mutable access
+//  We don't use `FrozenSet<T, ()>`, as the `StableDeref` requirements are different
```

Comment:
I think you meant `FrozenMap` here?

Reply:
whoops, accidentally hit this one when I was going through fixing comments
```suggestion
//  We don't use `FrozenMap<T, ()>`, as the `StableDeref` requirements are different
```